torch_em/data/datasets/light_microscopy/cvz_fluo.py

```diff
@@ -0,0 +1,167 @@
+and install using 'synapse get -r syn27624812'.
+
+The dataset is from the publication https://doi.org/10.1038/s41597-023-02108-z.
+Please cite it if you use this dataset for your research.
+"""  # noqa
+
+import os
+from glob import glob
+from tqdm import tqdm
+from pathlib import Path
+from natsort import natsorted
+from typing import Union, Literal, Tuple, Optional, List
+
+import imageio.v3 as imageio
+from skimage.measure import label as connected_components
+
+from torch.utils.data import Dataset, DataLoader
+
+import torch_em
+
+from .. import util
+from .neurips_cell_seg import to_rgb
+
+
+URL = "https://www.synapse.org/Synapse:syn27624812/"
+
+
+def get_cvz_fluo_data(path: Union[os.PathLike, str], download: bool = False):
+    """Obtain the CVZ-Fluo dataset.
+
+    Args:
+        path: Filepath to a folder where the downloaded data is saved.
+        download: Whether to download the data if it is not present.
+    """
+    if download:
+        raise NotImplementedError("Automatic download is not implemented for this dataset.")
+
+    data_dir = os.path.join(path, r"Annotation Panel Table.xlsx")
+    if os.path.exists(data_dir):
+        return
+    else:
+        raise AssertionError(f"The dataset is not downloaded. Please download it manually from '{URL}'.")
+
+
+def get_cvz_fluo_paths(
+    path: Union[os.PathLike, str],
+    stain_choice: Literal["cell", "dapi"],
+    data_choice: Optional[Literal["CODEX", "Vectra", "Zeiss"]] = None,
+    download: bool = False,
+) -> Tuple[List[str], List[str]]:
+    """Get paths to the CVZ-Fluo data.
+
+    Args:
+        path: Filepath to a folder where the downloaded data will be saved.
+        download: Whether to download the data if it is not present.
+
+    Returns:
+        List of filepaths for the image data.
+        List of filepaths for the label data.
+    """
+    get_cvz_fluo_data(path, download)
+
+    if data_choice is None:
+        data_choice = "**"
+    else:
+        if data_choice == "Zeiss" and stain_choice == "dapi":
+            raise ValueError("'Zeiss' data does not have DAPI stained images.")
+
+        data_choice = f"{data_choice}/**"
+
+    if stain_choice not in ["cell", "dapi"]:
+        raise ValueError(f"'{stain_choice}' is not a valid stain choice.")
+
+    raw_paths = natsorted(
+        glob(os.path.join(path, data_choice, f"*-Crop_{stain_choice.title()}_Png.png"), recursive=True)
+    )
+    label_paths = [p.replace("_Png.png", "_Mask_Png.png") for p in raw_paths]
+
+    neu_label_paths = []
+    for lpath in tqdm(label_paths, desc="Preprocessing labels"):
+        neu_lpath = lpath.replace(".png", ".tif")
+        neu_label_paths.append(neu_lpath)
+        if os.path.exists(neu_lpath):
+            continue
+
+        if not os.path.exists(lpath):  # HACK: some paths have weird spacing nomenclature.
+            lpath = Path(lpath).parent / rf" {os.path.basename(lpath)}"
```

Comment:
Why do we need a raw string here?

Reply:
There are a few labels which are named as `/path/to/ IMAGE.tif`, where there is a weird space infront of the capital "I". This creates issues and the path conventions cannot handle this apparently. So I just handle it by adding a space infront and allowing it as a raw string (due to the same reason as mentioned above)